test/models/test_vqvae.py

```diff
@@ -39,26 +68,10 @@ def test_data(self, decoder_input):
             quantized_flat=torch.tensor(
                 [[3.0, 9.0], [7.0, 21.0], [6.0, 12.0], [14.0, 28.0]]
             ),
-            codebook_indices=torch.tensor([3, 2, 1, 0]),
+            codebook_indices=torch.tensor([1, 3, 0, 2]),
```

Comment:
just confirming - this changed because random_seed is now autouse, and the codebook indices are the only thing depending on randomization?

Reply:
Right. In the previous version only `vqvae` gets the seed but `encoder` and `decoder` (containing `nn.Linear`) have changed the random state as well. Making seed `autouse` ensures that each layer inside `vqvae` gets a fresh random state.

To verify, I reverted this PR and added `seed` to `encoder` and `decoder` which replicated the change in this PR:

```
E       AssertionError: actual: tensor([1, 3, 0, 2]), expected: tensor([3, 2, 1, 0])

FAILED test/models/test_vqvae.py::TestVQVAE::test_encode - AssertionError: actual: tensor([1, 3, 0, 2]), expected: te...
FAILED test/models/test_vqvae.py::TestVQVAE::test_forward - AssertionError: actual: tensor([1, 3, 0, 2]), expected: t..
```
```
    @pytest.fixture(scope="class")
    def encoder(self, random):
        enc = nn.Linear(2, 2, bias=False)
        enc.weight = nn.Parameter(torch.tensor([[1.0, 2.0], [3.0, 4.0]]))
        return enc

    @pytest.fixture(scope="class")
    def decoder(self, random):
        dec = nn.Linear(2, 2, bias=False)
        dec.weight = nn.Parameter(torch.tensor([[1.0, 2.0], [3.0, 4.0]]))
        return dec

    @pytest.fixture(scope="class")
    def vqvae(self, encoder, decoder, random):
        return VQVAE(encoder, decoder, 4, 2)
```